qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/RunningSnapshotIT.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.upgrades;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.rest.ObjectPath;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+public class RunningSnapshotIT extends AbstractRollingUpgradeTestCase {
+
+    public RunningSnapshotIT(@Name("upgradedNodes") int upgradedNodes) {
+        super(upgradedNodes);
+    }
+
+    public void testRunningSnapshotCompleteAfterUpgrade() throws Exception {
+        final String indexName = "index";
+        final String repositoryName = "repo";
+        final String snapshotName = "snapshot";
+        final var nodeIds = getNodesInfo(client()).keySet();
+
+        if (isOldCluster()) {
+            // create an index to have one shard per node
+            createIndex(indexName, indexSettings(3, 0).put("index.routing.allocation.total_shards_per_node", 1).build());
+            ensureGreen(indexName);
```

Comment:
For the sake of completeness maybe (randomly) index some data?

Reply:
Sure see e211f26dafc96821be5da74d30881437166af666